### DISCUSSION TITLE
dnsdist: Support ACL per Server Pool?

### DISCUSSION THREAD

**Discussion by gshirazi** [PowerDNS/pdns#14660] [Ideas]
Currently, ACLs are applied globally: https://dnsdist.org/advanced/acl.html 

Would it be possible to support ACL per Server Pool e.g. support something like:

```
newServer({address="4.4.4.4", pool="default"})
newServer({address="1.1.1.1", pool="pool1"})
setACLForPool({'192.168.0.0/16'}, pool="pool1")
```
Then, in this example, if source IP in the request is `192.168.0.0/16`, query would go to `pool1` otherwise to `default` pool.

Or is this use-case already achievable with some lua script in `dnsdist.conf`? I did not find any Rule Selector for ACL matching https://dnsdist.org/reference/selectors.html

i.e. Is something like this possible? (sorry about syntax errors, not a lua expert, hope this conveys the concept): 
```
--- do we have any rule selector like matchACL
addAction(matchACLRule({'192.168.0.0/16'}), PoolAction("pool1"))
```

Thanks in advance for the help.

**Comment by omoerbeek** [comment]
There is https://dnsdist.org/reference/selectors.html#NetmaskGroupRule which can select on source or destination address of a query.

**Comment by gshirazi** [comment]
Cool thx @omoerbeek exactly what I wanted.
